camayoc/tests/qpc/cli/test_sources.py

```diff
@@ -286,6 +287,325 @@ def test_add_with_port_negative(
+    assert qpc_source_add.expect(pexpect.EOF) == 0
+    qpc_source_add.close()
+    assert qpc_source_add.exitstatus == 0
+
+    source_show_and_check(
+        {'name': name},
+        generate_show_output({
+            'cred_name': cred_name,
+            'hosts': hosts,
+            'name': name,
+            'options': {
+                'ssl_cert_verify': ssl_cert_verify.lower(),
+            },
+            'port': default_port_for_source(source_type),
+            'source_type': source_type,
+        })
+    )
+
+
+def test_add_with_ssl_cert_verify_negative(
+        isolated_filesystem, qpc_server_config, source_type):
+    """Try to add source with cred, hosts and an invalid ssl_cert_verify.
+
+    :id: ee108827-9695-4b6f-8c2b-bafb09f9ff85
+    :description: Add a source entry providing the ``--name``, ``--cred``,
+        ``--hosts`` and ``--ssl-cert-verify`` options. The value for the
+        ``--ssl-cert-verify`` should be an invalid value, for example a
+        different value from ``true`` or ``false`` for host manager types and
+        the option should not be valid for network type.
+    :steps: Run ``qpc source add --name <name> --cred <cred> --hosts <hosts>
+        --port <port> --ssl-cert-verify <invalid-ssl-cert-verify --type
+        <type>``
+    :expectedresults: An error message is printed and a non-zero status code
+        should be returned. Also no source entry is created.
+    """
+    cred_name = utils.uuid4()
+    hosts = '127.0.0.1'
+    name = utils.uuid4()
+    port = default_port_for_source(source_type)
+    if source_type == 'network':
+        ssl_cert_verify = random.choice(QPC_BOOLEAN_VALUES)
+        expected_error = (
+            'Error: Invalid SSL options for network source: ssl_cert_verify')
+        exitstatus = 1
+    else:
+        ssl_cert_verify = utils.uuid4()
+        expected_error = (
+            'qpc source add: error: argument --ssl-cert-verify: invalid '
+            'choice: \'{}\' \\(choose from \'True\', \'False\', \'true\', '
+            '\'false\'\\)'
+            .format(ssl_cert_verify)
+        )
+        exitstatus = 2
+    cred_add_and_check(
+        {
+            'name': cred_name,
+            'username': utils.uuid4(),
+            'password': None,
+            'type': source_type,
+        },
+        [(CONNECTION_PASSWORD_INPUT, utils.uuid4())],
+    )
+
+    qpc_source_add = pexpect.spawn(
+        'qpc source add --name {} --cred {} --hosts {} --port {} '
+        '--ssl-cert-verify {} --type {}'
+        .format(name, cred_name, hosts, port, ssl_cert_verify, source_type)
+    )
+    assert qpc_source_add.expect(expected_error) == 0
+    assert qpc_source_add.expect(pexpect.EOF) == 0
+    qpc_source_add.close()
+    assert qpc_source_add.exitstatus == exitstatus
+
+
+@pytest.mark.parametrize('source_type', QPC_HOST_MANAGER_TYPES)
+@pytest.mark.parametrize('ssl_protocol', QPC_SSL_PROTOCOL_VALUES)
```

Comment:
DId not realize this was an option to set the protocol. We don't have any API tests for this.